tests/unit/replication/ReplicateObject.spec.js

```diff
@@ -0,0 +1,131 @@
+                    destConfig: config.extensions.replication.destination,
+                    site: 'test-site-2',
+                    notificationConfig,
+                }),
+            });
+            const publishNotificationStub = sinon.stub(task, '_publishFailedReplicationStatusNotification');
+            const publishReplicationStatusStub = sinon.stub(task, '_publishReplicationStatus');
+            task._handleReplicationOutcome(null, sourceEntry, sourceEntry, replicationEntry, fakeLogger,
+                (err, data) => {
+                assert.ifError(err);
+                assert(data);
+                assert(publishReplicationStatusStub.calledOnceWith(sourceEntry, 'COMPLETED', {
+                    log: fakeLogger,
+                    kafkaEntry: replicationEntry,
+                }));
+                assert(publishNotificationStub.notCalled);
+                return done();
+            });
+        });
+    });
+
+    describe('::_publishFailedReplicationStatusNotification', () => {
+        it('should publish entry to notification topic in correct format', done => {
+            sinon.stub(configUtil, 'validateEntry').returns(true);
+            const notificationProducerSend = sinon.stub().callsFake((entries, cb) => cb());
+            const task = new ReplicateObject({
+                getStateVars: () => ({
+                    repConfig: {
+                        queueProcessor: {
+                            retry: retryConfig,
+                        },
+                    },
+                    sourceConfig: config.extensions.replication.source,
+                    destConfig: config.extensions.replication.destination,
+                    site: 'test-site-2',
+                    notificationConfig,
+                    notificationProducer: {
+                        send: notificationProducerSend,
+                    },
+                    notificationConfigManager: {
+                        getConfig: () => ({
+                            key: 'value',
+                        }),
+                    }
+                }),
+            });
+            sinon.stub(sourceEntry, 'getObjectVersionedKey').returns('example-key\u0000123456');
+            sinon.stub(sourceEntry, 'getVersionId').returns('123456');
+            sinon.stub(sourceEntry, 'getBucket').returns('example-bucket');
+            sinon.stub(sourceEntry, 'getValue').returns({
+                'last-modified': '0000',
+                'originOp': 'origin-operation',
+                'dataStoreName': 'metastore',
+                'md-model-version': '1',
+                'content-length': '6',
+                'versionId': '1234',
+            });
+            task._publishFailedReplicationStatusNotification(sourceEntry, fakeLogger);
+            const entryPublished = {
+                key: 'example-bucket',
+                message: '{\"dateTime\":\"0000\",\"eventType\":\"s3:Replication:OperationFailedReplication\"' +
```

Comment:
Not sure you need to escape the double quotes here. This may also be more readable as a `JSON.stringify`d object, even if it feels too close to implementation? Up to you